File: Python/8py 1/guess_blue.py
"""
|--------------------------------------------------------------------------
| Thinkful - Number Drills: Blue and red marbles
|--------------------------------------------------------------------------
|
"""

def guess_blue(bs, rs, bp, rp):
   return (bs-bp) / ((bs-bp)+(rs-rp))


guess_blue = lambda bs, rs, bp, rp: (bs-bp)/(bs+rs-bp-rp)


def guess_blue(blue_start, red_start, blue_pulled, red_pulled):
   remaining_blue = blue_start - blue_pulled
   remaining_red = red_start - red_pulled
   total_remaining = remaining_blue + remaining_red
   return remaining_blue / total_remaining


print(guess_blue(5, 5, 2, 3), 0.6)
print(guess_blue(5, 7, 4, 3), 0.2)
print(guess_blue(12, 18, 4, 6), 0.4)

File: JavaScript/8js 2/guessBlue.js
/*
|--------------------------------------------------------------------------
| Thinkful - Number Drills: Blue and red marbles
|--------------------------------------------------------------------------
|
*/

function guessBlue(bs, rs, bp, rp) {
   return (bs - bp) / (bs - bp + (rs - rp));
}


const guessBlue = (a1, b1, a2, b2) => (a1 = a1 - a2) / (a1 + b1 - b2);


function guessBlue(blueStart, redStart, bluePulled, redPulled) {
   let leftBlue = blueStart - bluePulled;
   let leftRed = redStart - redPulled;
   let leftTotal = leftBlue + leftRed;
   return leftBlue / leftTotal;
}


console.log(guessBlue(5, 5, 2, 3), 0.6);
console.log(guessBlue(5, 7, 4, 3), 0.2);
console.log(guessBlue(12, 18, 4, 6), 0.4);